infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -720,6 +720,40 @@ public void processTopologyChangeNotification(DiscoveryServiceEvent event) {
         }
     }
 
+    /**
+     * Determine if any standby cluster has been added or removed from the topology
+     *
+     * @param discoveredTopology new discovered topology
+     * @return true, standby has been added or removed
+     * false, otherwise
+     */
+    private boolean isStandbyChange(TopologyDescriptor discoveredTopology) {
+        return topologyDescriptor.getStandbyClusters().keySet() != discoveredTopology.getStandbyClusters().keySet() ||
+                isNodeAddedOrRemoved(discoveredTopology.getStandbyClusters(), topologyDescriptor.getStandbyClusters());
+    }
+
+    /**
+     * Determine if any active cluster has been added or removed from the topology
+     *
+     * @param discoveredTopology new discovered topology
+     * @return true, standby has been added or removed
+     * false, otherwise
+     */
+    private boolean isActiveChange(TopologyDescriptor discoveredTopology) {
+        return topologyDescriptor.getActiveClusters().keySet() != discoveredTopology.getActiveClusters().keySet() ||
+                isNodeAddedOrRemoved(discoveredTopology.getActiveClusters(), topologyDescriptor.getActiveClusters());
+    }
+
+    private boolean isNodeAddedOrRemoved(Map<String, ClusterDescriptor> newClusterIdToDescriptor,
+                                         Map<String, ClusterDescriptor> oldClusterIdToDescriptor) {
```

Comment:
nit -> I would not call it 'old' but 'current' instead, as 'new' might not be applied, which means there is no such thing as old, other than the actual current one.